### PR TITLE
[Merged by Bors] - feat(GroupTheory/Index): index of product of subgroups

### PR DESCRIPTION
**Pull request by jsm28**
Add two lemmas about the index of a product (pairwise or indexed) of subgroups.

```lean
@[to_additive]
lemma index_prod (H : Subgroup G) (K : Subgroup G') : (H.prod K).index = H.index * K.index := by
```

```lean
@[to_additive]
lemma index_pi {ι : Type*} [Fintype ι] (H : ι → Subgroup G) :
    (Subgroup.pi Set.univ H).index = ∏ i, (H i).index := by
```

From AperiodicMonotilesLean.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by tb65536** [comment]
maintainer merge

File: Mathlib/GroupTheory/Index.lean [hunk 1]
```diff
@@ -448,6 +448,21 @@ lemma pow_mem_of_relindex_ne_zero_of_dvd (h : H.relindex K ≠ 0) {a : G} (ha :
   convert pow_mem_of_index_ne_zero_of_dvd h ⟨a, ha⟩ hn
   simp [pow_mem ha, mem_subgroupOf]
 
+@[to_additive]
+lemma index_prod (H : Subgroup G) (K : Subgroup G') : (H.prod K).index = H.index * K.index := by
```
Comment (jcommelin) [inline Mathlib/GroupTheory/Index.lean:452]: Should this be a simp lemma? Same question for below?

Comment (jsm28) [reply Mathlib/GroupTheory/Index.lean:452]: I've made both into simp lemmas.

**Review by Vierkantor — APPROVED** [review]
bors merge

**Pull request closed without merging**